habitat/tasks/eqa/eqa_task.py

```diff
@@ -48,77 +46,225 @@ class EQAEpisode(NavigationEpisode):
     )
 
 
+@registry.register_sensor
 class QuestionSensor(Sensor):
-    def __init__(self, **kwargs):
+    def __init__(self, dataset, **kwargs):
         self.uuid = "question"
         self.sensor_type = SensorTypes.TEXT
-        # TODO (maksymets) extend gym observation space for text and metadata
-        self.observation_space = spaces.Discrete(0)
+        self._dataset = dataset
+
+        self.observation_space = spaces.Discrete(
+            len(dataset.get_questions_vocabulary())
+        )
 
-    def _get_observation(
+    def get_observation(
         self,
         observations: Dict[str, Observations],
         episode: EQAEpisode,
-        **kwargs
+        **kwargs,
     ):
-        return episode.question.question_text
+        return self._dataset.get_questions_vocabulary()[
```

Comment:
This'll need to be different.  Needs to return a list of token IDs
```suggestion
        return [self._dataset.get_questions_vocabulary()[tk] for tk in episode.question.question_tokens]
```